core/domain/caching_services.py

```diff
@@ -0,0 +1,256 @@
+# Exploration. If we are not storing multiple versions of an Exploration in the
+# cache, the sub-namespace is not required to differentiate the keys of multiple
+# versions of an exploration and can therefore be None.
+CACHE_NAMESPACE_EXPLORATION = 'exploration'
+# This namespace supports sub-namespaces which are identified by the stringified
+# version number of the collections within the sub-namespace. The value for
+# each key in this namespace should be a serialized representation of a
+# Collection. If we are not storing multiple versions of a Collection in the
+# cache, the sub-namespace is not required to differentiate the keys of multiple
+# versions of a collection and can therefore be None.
+CACHE_NAMESPACE_COLLECTION = 'collection'
+# This namespace supports sub-namespaces which are identified by the stringified
+# version number of the skills within the sub-namespace. The value for
+# each key in this namespace should be a serialized representation of a
+# Skill. If we are not storing multiple versions of a Skill in the
+# cache, the sub-namespace is not required to differentiate the keys of multiple
+# versions of a skill and can therefore be None.
+CACHE_NAMESPACE_SKILL = 'skill'
+# This namespace supports sub-namespaces which are identified by the stringified
+# version number of the stories within the sub-namespace. The value for
+# each key in this namespace should be a serialized representation of a
+# Story. If we are not storing multiple versions of a Story in the
+# cache, the sub-namespace is not required to differentiate the keys of multiple
+# versions of a story and can therefore be None.
+CACHE_NAMESPACE_STORY = 'story'
+# This namespace supports sub-namespaces which are identified by the stringified
+# version number of the topics within the sub-namespace. The value for
+# each key in this namespace should be a serialized representation of a
+# Topic. If we are not storing multiple versions of a Topic in the
+# cache, the sub-namespace is not required to differentiate the keys of multiple
+# versions of a topic and can therefore be None.
+CACHE_NAMESPACE_TOPIC = 'topic'
+# The value for each key in this namespace should be a serialized representation
+# of a ConfigPropertyModel value (the 'value' attribute of a ConfigPropertyModel
+# object). Since we are not storing multiple versions of a ConfigPropertyModel,
+# the sub-namespace is not required to differentiate the keys of multiple
+# versions of a ConfigPropertyModel and can therefore be None.
+CACHE_NAMESPACE_CONFIG = 'config'
+# The sub-namespace is not necessary for the default namespace. The namespace
+# handles default datatypes allowed by Redis including Strings, Lists, Sets,
+# and Hashes. More details can be found at: https://redis.io/topics/data-types.
+CACHE_NAMESPACE_DEFAULT = 'default'
+
+DESERIALIZATION_FUNCTIONS = {
+    CACHE_NAMESPACE_COLLECTION: collection_domain.Collection.deserialize,
```

Comment:
Pull ':' out into a top-level MEMCACHE_KEY_DELIMITER constant and use it everywhere.

Have a backend test that ensures that every key in this dict does not contain that constant.

Reply:
Done.